presto-main/src/test/java/io/prestosql/execution/buffer/BenchmarkDataGenerator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution.buffer;
+
+import io.prestosql.spi.type.LongTimestamp;
+import io.prestosql.spi.type.SqlDecimal;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Random;
+import java.util.function.Function;
+
+import static io.prestosql.spi.type.Timestamps.PICOSECONDS_PER_MICROSECOND;
+
+public class BenchmarkDataGenerator
+{
+    public static final int MAX_STRING = 19;
+    public static final int LONG_DECIMAL_PRECISION = 30;
+    public static final int LONG_DECIMAL_SCALE = 5;
+
+    private BenchmarkDataGenerator()
+    {
+    }
+
+    public static <T> Iterator<T> createValues(int size, Function<Random, T> valueGenerator, double nullChance)
+    {
+        Random random = new Random(0);
+        List<T> values = new ArrayList<>();
+        for (int i = 0; i < size; ++i) {
+            if (random.nextDouble() >= nullChance) {
```

Comment:
Although extremely unlikely (1 in 2^54?), `random.nextDouble()` can return 0 which means that `nullChange = 0` can produce a null value.